dashboard/app/controllers/application_controller.rb

```diff
@@ -154,6 +154,12 @@ def milestone_response(options)
       end
     end
 
+    if PuzzleRating.enabled?
+      if script_level && PuzzleRating.can_rate?(script_level.script, script_level.level, current_user)
```

Comment:
oh, I see, it's not just permissions, it's whether you've seen the puzzle before.